src/server/scripts/EasternKingdoms/Karazhan/boss_tenris_mirkblood.cpp

```diff
@@ -0,0 +1,272 @@
+    SPELL_DASH_GASH_RETURN_TO_TANK                = 50924, // charge
+    SPELL_DASH_GASH_RETURN_TO_TANK_PRE_SPELL      = 50925, // null
+    SPELL_DASH_GASH_RETURN_TO_TANK_PRE_SPELL_ROOT = 50932, // null
+
+    SPELL_DESPAWN_SANGUINE_SPIRIT_VISUAL             = 51214, // dummy
+    SPELL_DESPAWN_SANGUINE_SPIRITS                   = 51212, // script, dummy
+    SPELL_SANGUINE_SPIRIT_AURA                       = 50993, // dummy, periodic trigger 51013
+    SPELL_SANGUINE_SPIRIT_PRE_AURA                   = 51282, // dummy
+    SPELL_SANGUINE_SPIRIT_PRE_AURA2                  = 51283, // size mod
+    SPELL_SUMMON_SANGUINE_SPIRIT0                    = 50996, // null
+    SPELL_SUMMON_SANGUINE_SPIRIT1                    = 50998, // trigger missile 50996
+    SPELL_SUMMON_SANGUINE_SPIRIT2                    = 51204, // trigger missile 50996
+    SPELL_SUMMON_SANGUINE_SPIRIT_MISSILE_BURST       = 51208, // periodic trigger 50998
+    SPELL_SUMMON_SANGUINE_SPIRIT_SHORT_MISSILE_BURST = 51280, // periodic trigger 50998
+    SPELL_SUMMON_SANGUINE_SPIRIT_ON_KILL             = 51205, // dummy
+    SPELL_EXSANGUINATE                               = 51013, // damage
+};
+
+struct boss_tenris_mirkblood : public BossAI
+{
+    boss_tenris_mirkblood(Creature* creature) : BossAI(creature, DATA_MIRKBLOOD)
+    {
+        scheduler.SetValidator([this]
+            {
+                return !me->HasUnitState(UNIT_STATE_CASTING);
+            });
+    }
+
+    void Reset() override
+    {
+        _Reset();
+
+        ScheduleHealthCheckEvent(50, [&] {
+            Talk(SAY_SUMMON);
+            DoCast(SPELL_SUMMON_SANGUINE_SPIRIT_MISSILE_BURST);
+            });
+
+        ScheduleHealthCheckEvent(45, [&] {
+            scheduler.Schedule(10s, 15s, [this](TaskContext context)
+                {
+                    DoCast(SPELL_BLOOD_TAP);
+                    context.Repeat(15s, 40s);
+                });
+            });
+
+        scheduler.CancelAll();
+    }
+
+    void JustEngagedWith(Unit* /*who*/) override
+    {
+        Talk(SAY_AGGRO);
+        DoZoneInCombat();
+
+        scheduler.Schedule(1s, 5s, [this](TaskContext context)
+            { // Blood Mirror
+                DoCast(SPELL_BLOOD_MIRROR_TARGET_PICKER);
+                context.Repeat(20s, 50s);
+            }).Schedule(30s, [this](TaskContext context)
+            { // Blood Swoop
+                DoCast(SPELL_DASH_GASH_PRE_SPELL);
+                context.Repeat(15s, 40s);
+            }).Schedule(6s, 15s, [this](TaskContext context)
+            {
+                DoCast(SPELL_SUMMON_SANGUINE_SPIRIT_SHORT_MISSILE_BURST);
+                context.Repeat(6s, 15s);
+            });
+    }
+
+    void JustSummoned(Creature* summoned) override
+    {
+    }
+
+    void KilledUnit(Unit* victim) override
+    {
+    }
+
+    void DamageTaken(Unit* /*attacker*/, uint32& damage, DamageEffectType /*damagetype*/, SpellSchoolMask /*damageSchoolMask*/) override
```

Comment:
Need to explicitely call the hook from the parent class inside this else the scheduled health event won't fire

BossAI:DamageTaken(...)

Reply:
Should be corrected.